### DISCUSSION TITLE
BUILD FAILED (macOS 14.6.1 on arm64 using ruby-build 20240917-2-ga22ff72e)

### DISCUSSION THREAD

**Discussion by sglien** [rbenv/ruby-build#2449] [Build failures]
It seems that I'm encountering a build failure when attempting to install Ruby, even with OpenSSL 3 and other dependencies installed. Not sure how to fix it. I referred to a few similar problems here such as: 

https://github.com/rbenv/ruby-build/discussions/2439 

https://github.com/rbenv/ruby-build/discussions/2404

but it does not seem to work for my problem. Maybe I'm missing somethng? 

This is the main issue i was trying to solve:

ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /Library/Ruby/Gems/2.6.0 directory. 

System: Apple M2. Sonoma 14.6.1 (23G93)

Command: rbenv install 3.1.2

ruby-build: using openssl@3 from homebrew
==> Installing ruby-3.1.2...
ruby-build: using readline from homebrew
ruby-build: using libyaml from homebrew
ruby-build: using gmp from homebrew
-> ./configure "--prefix=$HOME/.rbenv/versions/3.1.2" --with-openssl-dir=/usr/local/opt/openssl@3 --enable-shared --with-readline-dir=/usr/local/opt/readline --with-libyaml-dir=/usr/local/opt/libyaml --with-gmp-dir=/usr/local/opt/gmp --with-ext=openssl,psych,+
-> make -j 8

BUILD FAILED (macOS 14.6.1 on arm64 using ruby-build 20240917-2-ga22ff72e)

You can inspect the build directory at /var/folders/21/7tzwqsj13cncth16rxwtlgrm0000gn/T/ruby-build.20240929131438.15790.Jlublk
See the full build log at /var/folders/21/7tzwqsj13cncth16rxwtlgrm0000gn/T/ruby-build.20240929131438.15790.log

**Comment by mislav** [comment]
Hi, I'm sorry you're having trouble installing a Ruby version.

> ERROR: While executing gem ... (Gem::FilePermissionError)
> You don't have write permissions for the /Library/Ruby/Gems/2.6.0 directory.

This is very strange. It seems like something is trying to install a gem into the location of the system Ruby, but I would never expect this during an installation of a Ruby version. Did you get this error message from the log file that was the result of the `rbenv install` command? Could you paste more of the log, perhaps as a GitHub Gist?

> Command: rbenv install 3.1.2

Just to check something, could you try to install the latest Ruby 3.1.x version, which is at the time of writing `rbenv install 3.1.6`? Maybe there are bugs in the build system of 3.1.2 that got fixed in newer versions. Thanks
